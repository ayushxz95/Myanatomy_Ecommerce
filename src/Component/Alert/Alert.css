.alert-container{
    background-color: rgba(250, 154, 194, 0.185);
}