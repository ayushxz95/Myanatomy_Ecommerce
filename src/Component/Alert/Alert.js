import * as React from 'react';
import { Component } from "react";
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import { PropTypes } from "prop-types";




class Alert extends Component {

  constructor(props){
      super(props);
      this.state = {
          open: false,
      }
  }

  handleClick = () => {
    this.setState((prev) => ({ open: !prev.open }));
  };

  onSubmit = (e) => {
    if (this.props.Alertmessage === "Update") {
      this.props.updateItem(this.props.state, this.props.id, this.props.type, this.props.category);
      this.handleClick();
      this.props.handleClick();
      this.props.submit();
    } else if(this.props.Alertmessage === "Add"){
      this.props.addItems(this.props.state, this.props.category);
      this.handleClick();
      this.props.handleClick();
      this.props.AddSubmit();
    } else {
      this.handleClick();
      this.props.handleClick();
      this.props.submit();
    }
  };

     render(){
       return (
        <div>
        <Button color={this.props.color} onClick={this.handleClick} disabled={this.props.disabled}>
          {this.props.Alertmessage}
        </Button>
        <div className="alert-container">
        <Dialog open={this.state.open} onClose={this.handleClick}>
          <DialogTitle>Confirm</DialogTitle>
          <DialogContent>
            <DialogContentText>
              {this.props.Alertmessage === "Update" || this.props.Alertmessage === "Add" ? `Are you sure ? You want to save the changes ?` : `Are you sure ? You don't want to save  the changes ?`}
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClick} color="error">Cancel</Button>
            <Button id="btn-alert" onClick={this.onSubmit} color="success">Sure</Button>
          </DialogActions>
         
        </Dialog>
        </div>
      </div>
        )
     }
}
Alert.propTypes = {
  updateItem: PropTypes.func,
  addItems: PropTypes.func,
  fetchItems: PropTypes.func,
  fetchCategoryItems: PropTypes.func,
}
  
export default Alert;