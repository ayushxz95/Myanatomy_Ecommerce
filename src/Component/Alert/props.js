import { addItems, fetchItems, updateItem, fetchCategoryItemsSuccess } from "../../Redux/Actions/ItemAction";

const mapStateToProps = (state) => {
  return {
    category: state.category.category,
  }
}

const mapDispatchToProps = (dispatch) => {
    return {
      updateItem: (state, id, type, category) => dispatch(updateItem(state, id, type, category)),
      addItems: (state, type) => dispatch(addItems(state, type)),
      fetchItems: () => dispatch(fetchItems()),
      fetchCategoryItems: (type) => dispatch(fetchCategoryItemsSuccess(type)),
    };
  };

export {mapStateToProps, mapDispatchToProps};