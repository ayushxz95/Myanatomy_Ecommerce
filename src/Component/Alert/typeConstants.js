export const mockItem = {
      state:5,
      id:1,
      type:"Electronics",
      statetype:"Electronics",
      Alertmessage:"Cancel",
      color:"error",
  };