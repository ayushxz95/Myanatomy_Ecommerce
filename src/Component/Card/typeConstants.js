export const mockItem = {
    name: "Fan",
    id: 1,
    type: "Electronics",
    company: "Havells",
    key: 1,
  };