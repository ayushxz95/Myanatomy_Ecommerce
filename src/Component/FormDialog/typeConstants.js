export const mockItem = {
  message: "Update Product",
  buttonmessage: "Update",
  type: "Electronics",
  company: "Bajaj",
  name: "Fan",
  id: 3,
};
