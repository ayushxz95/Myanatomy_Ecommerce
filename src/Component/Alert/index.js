import { connect } from "react-redux";
import Alert from "./Alert";
import {mapStateToProps , mapDispatchToProps} from "./props";

export default connect(mapStateToProps, mapDispatchToProps)(Alert);