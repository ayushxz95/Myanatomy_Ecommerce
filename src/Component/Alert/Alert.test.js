import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Alert from "./Alert";
import {mockItem} from "./typeConstants"

configure({ adapter: new Adapter() });


describe("Testing Alert Component", () => {
  it("Matching Alert's snapshot", () => {
    const handleClickmock = jest.fn();
    const onSubmitmock = jest.fn();
    const onAddSubmitmock = jest.fn();
    const component = shallow(
      <Alert
      state={mockItem.state}
      id={mockItem.id}
      type={mockItem.type}
      statetype={mockItem.type}
      Alertmessage={mockItem.Alertmessage}
      handleClick={handleClickmock}
      color={mockItem.color}
      submit={onSubmitmock}
      AddSubmit={onAddSubmitmock}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("Button is Clickable", () => {
    const handleClickmock = jest.fn();
    const onSubmitmock = jest.fn();
    const onAddSubmitmock = jest.fn();
    const updateItemmock = jest.fn()
    const component = shallow(
      <Alert
      state={mockItem.state}
      id={mockItem.id}
      type={mockItem.type}
      statetype={mockItem.type}
      Alertmessage={mockItem.Alertmessage}
      handleClick={handleClickmock}
      color={mockItem.color}
      submit={onSubmitmock}
      AddSubmit={onAddSubmitmock}
      />
    );
    const instance = component.instance();
    console.log(instance);
    jest.spyOn(instance, "onSubmit"); 
    const updateButton = component.find("#btn-alert");
    updateButton.simulate("click");
    expect(updateItemmock).toHaveBeenCalledTimes(1);
  });
});