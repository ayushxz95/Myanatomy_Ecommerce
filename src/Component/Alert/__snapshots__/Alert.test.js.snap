// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Alert Component Matching Alert's snapshot 1`] = `
<div>
  <ForwardRef(Button)
    color="error"
    onClick={[Function]}
  >
    Cancel
  </ForwardRef(Button)>
  <div
    className="alert-container"
  >
    <ForwardRef(Dialog)
      onClose={[Function]}
      open={false}
    >
      <ForwardRef(DialogTitle)>
        Confirm
      </ForwardRef(DialogTitle)>
      <ForwardRef(DialogContent)>
        <ForwardRef(DialogContentText)>
          Are you sure ? You don't want to save  the changes ?
        </ForwardRef(DialogContentText)>
      </ForwardRef(DialogContent)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button)
          color="error"
          onClick={[Function]}
        >
          Cancel
        </ForwardRef(Button)>
        <ForwardRef(Button)
          color="success"
          id="btn-alert"
          onClick={[Function]}
        >
          Sure
        </ForwardRef(Button)>
      </ForwardRef(DialogActions)>
    </ForwardRef(Dialog)>
  </div>
</div>
`;
